tests/m2m_regress/tests.py

```diff
@@ -71,6 +71,20 @@ def test_m2m_inheritance_symmetry(self):
         self.assertSequenceEqual(sr_child.related.all(), [sr_sibling.selfrefer_ptr])
         self.assertSequenceEqual(sr_sibling.related.all(), [sr_child.selfrefer_ptr])
 
+    def test_m2m_instance_clone(self):
+        t1 = Tag.objects.create(name="t1")
+        Entry.objects.create(name="e1")
+        entry = Entry.objects.all()[0]
+        entry.topics.set([t1])
+        old_topics = entry.topics.all()
+        entry.pk = None
+        entry._state.adding = True
```

Comment:
It worked in Django 4.0 without setting `_state.adding`. IMO we shouldn't inspect or use `_state.adding` in tests and a potential fix.